test/gtest/tensor_2d_gen_ocl_hip.cpp

```diff
@@ -0,0 +1,426 @@
+        {
+            for(size_t N = 1; N <= maxTotalSize; N *= 2)
+            {
+                for(size_t C = 1; C <= maxTotalSize / N; C *= 2)
+                {
+                    size_t totalSize = N * C;
+                    // Ensure the total size does not exceed the maximum limit
+                    if(totalSize <= maxTotalSize)
+                    {
+                        configs.push_back({{N, C}, {C, 1}, {N, C}, {C, 1}});
+                        configs.push_back({{N, C}, {C, 1}, {1, C}, {C, 1}});
+                        configs.push_back({{N, C}, {C, 1}, {N, 1}, {1, 1}});
+                        configs.push_back({{N, C}, {C, 1}, {1, 1}, {1, 1}});
+                    }
+                }
+            }
+        }
+        else
+        {
+            for(size_t N = 2; N <= maxTotalSize; N *= 2)
+            {
+                for(size_t C = 2; C <= maxTotalSize / N; C *= 2)
+                {
+                    size_t totalSize = (N - 1) * (C - 1);
+                    // Ensure the total size does not exceed the maximum limit
+                    if(totalSize <= maxTotalSize)
+                    {
+                        configs.push_back({{N - 1, C - 1}, {C - 1, 1}, {N - 1, C - 1}, {C - 1, 1}});
+                        configs.push_back({{N - 1, C - 1}, {C - 1, 1}, {1, C - 1}, {C - 1, 1}});
+                        configs.push_back({{N - 1, C - 1}, {C - 1, 1}, {N - 1, 1}, {1, 1}});
+                        configs.push_back({{N - 1, C - 1}, {C - 1, 1}, {1, 1}, {1, 1}});
+                    }
+
+                    totalSize = (N + 1) * (C + 1);
+                    if(totalSize <= maxTotalSize)
+                    {
+                        configs.push_back({{N + 1, C + 1}, {C + 1, 1}, {N + 1, C + 1}, {C + 1, 1}});
+                        configs.push_back({{N + 1, C + 1}, {C + 1, 1}, {1, C + 1}, {C + 1, 1}});
+                        configs.push_back({{N + 1, C + 1}, {C + 1, 1}, {N + 1, 1}, {1, 1}});
+                        configs.push_back({{N + 1, C + 1}, {C + 1, 1}, {1, 1}, {1, 1}});
+                    }
+                }
+            }
+        }
+
+        return configs;
+    }
+    else
+    {
+        return {{{16, 8}, {8, 1}, {16, 8}, {8, 1}},
+                {{16, 8}, {8, 1}, {1, 8}, {8, 1}},
+                {{16, 8}, {8, 1}, {16, 1}, {1, 1}},
+                {{16, 8}, {8, 1}, {1, 1}, {1, 1}},
+                {{20, 16}, {16, 1}, {20, 16}, {16, 1}},
+                {{20, 16}, {16, 1}, {1, 16}, {16, 1}},
+                {{20, 16}, {16, 1}, {20, 1}, {1, 1}},
+                {{20, 16}, {16, 1}, {1, 1}, {1, 1}},
+                {{32, 64}, {64, 1}, {32, 64}, {64, 1}},
+                {{32, 64}, {64, 1}, {1, 64}, {64, 1}},
+                {{32, 64}, {64, 1}, {32, 1}, {1, 1}},
+                {{32, 64}, {64, 1}, {1, 1}, {1, 1}}};
+    }
+}
+
+template <typename T>
+struct Op2DTensorGenericTest
+    : public ::testing::TestWithParam<std::tuple<TensorsConfig, std::vector<float>>>
```

Comment:
For the sake of performance, it's better to use static datatypes and replace `std::vector<flaot>` by `std::array<float,3>` or `struct AlphaBeta{};` or even 3 independent `float` values and you can generate testcases as simple as
```C++
 testing::Combine(testing::ValuesIn(TensorsConfigs<float>()),
                                          testing::Values(1.0f),
                                          testing::Values(1.0f),
                                          testing::Values(0.0f, 1.0f))
```
which will generate the cases {1,1,0} and {1,1,1}.

It's just for the information and possibilities.